waku/waku_rln_relay/rln_relay.nim

```diff
@@ -103,23 +105,24 @@ proc stop*(rlnPeer: WakuRLNRelay) {.async: (raises: [Exception]).} =
   await rlnPeer.groupManager.stop()
 
 proc hasDuplicate*(rlnPeer: WakuRLNRelay,
+                   epoch: Epoch,
                    proofMetadata: ProofMetadata): RlnRelayResult[bool] =
   ## returns true if there is another message in the  `nullifierLog` of the `rlnPeer` with the same
   ## epoch and nullifier as `proofMetadata`'s epoch and nullifier
   ## otherwise, returns false
   ## Returns an error if it cannot check for duplicates
 
-  let externalNullifier = proofMetadata.externalNullifier
   # check if the epoch exists
-  if not rlnPeer.nullifierLog.hasKey(externalNullifier):
```

Comment:
geez, don't know how this change got into master. without the change for the epoch check instead of externalNullifier check, a waku node would accept only one message per epoch

Reply:
wot? seems that this has been like this for quite some time. but i haven't observed that behaviour when testing with waku-simulator a while back.

in any case, if its that important, can we put into a different PR?. also so that the next incoming release has this fix.

Reply:
were there multiple publishers or just a single one in those simulations?

Reply:
also, while i agree with you that these changes are important, this PR is ready for review and checks are green - hope it's okay that i don't split the PRs?

Reply:
mm imho its not only about getting that change into master, but also about having PRs addressing specific things. in this case this is addressing a major bug, so its not a "testing + improvements". It is also beneficial to have such bug as an isolated PR, so that its not left unnoticed, its also seen in commit history and I'm sure the one preparing the next release would appreciate it.

effort should be quite low right? its just 1 line of code?

so i would suggest a separate PR.

Reply:
> were there multiple publishers or just a single one in those simulations?

iirc simulator.waku.org had multiple ones, and i could see in the network more than 1 msg/sec published. unfortunately deployment seems down now.

Reply:
cool, will let this stay in draft and create a new pr for the nullifierLog change tomorrow

Reply:
waiting for checks to pass, but https://github.com/waku-org/nwaku/pull/2508